code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,307 @@
+/obj/item/inflatable
+	name = "inflatable wall"
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation."
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "folded_wall"
+	w_class = 3
+
+/obj/item/inflatable/attack_self(mob/user)
+	playsound(loc, 'sound/items/zip.ogg', 75, 1)
+	user << "<span class='notice'>You inflate [src].</span>"
+	var/obj/structure/inflatable/R = new /obj/structure/inflatable(user.loc)
+	src.transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	qdel(src)
+
+/obj/structure/inflatable
+	name = "inflatable wall"
+	desc = "An inflated membrane. Do not puncture."
+	density = 1
+	anchored = 1
+	opacity = 0
+
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "wall"
+
+	var/health = 50.0
+
+/obj/structure/inflatable/New(location)
+	..()
+	air_update_turf(1)
+
+/obj/structure/inflatable/Destroy()
+	air_update_turf(1)
+	return ..()
+
+/obj/structure/inflatable/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	return 0
+
+/obj/structure/inflatable/CanAtmosPass(turf/T)
+	return !density
+
+/obj/structure/inflatable/bullet_act(var/obj/item/projectile/Proj)
+	health -= Proj.damage
+	..()
```

Comment:
Just checked. There's no definition. I was worried you might accidentally do twice much damage